src/Language/PureScript/Parser/Lexer.hs

```diff
@@ -213,8 +216,10 @@ parseToken = P.choice
   , P.try $ P.char '_'    *> P.notFollowedBy identLetter *> pure Underscore
   , HoleLit <$> P.try (P.char '?' *> P.many1 identLetter)
   , LName         <$> parseLName
-  , do uName <- parseUName
-       (guard (validModuleName uName) >> Qualifier uName <$ P.char '.') <|> pure (UName uName)
+  , parseUName >>= \uName ->
+      (guard (validModuleName uName) >> Qualifier uName <$ P.char '.')
+        <|> (guard (validUName uName) >> pure (UName uName))
```

Comment:
Actually, can we? The `P.char '.'` thing in there for `Qualifier` probably obstructs that. Should work for `uname` / `tyname` though.